velocity_modelling/cvm/write/csv.py

```diff
@@ -0,0 +1,104 @@
+"""
+Module for writing velocity model data to CSV format.
+"""
+
+import csv
+from pathlib import Path
+
+from velocity_modelling.cvm.geometry import (
+    PartialGlobalMesh,
+)
+from velocity_modelling.cvm.logging import VMLogger
+from velocity_modelling.cvm.velocity3d import (
+    PartialGlobalQualities,
+)
+
+
+def write_global_qualities(
+    output_dir: Path,
+    partial_global_mesh: PartialGlobalMesh,
+    partial_global_qualities: PartialGlobalQualities,
+    lat_ind: int,
+    min_vs: float = 0.0,
+    logger: VMLogger = None,
+):
+    """
+    Write the velocity model to CSV format.
+
+    Parameters
+    ----------
+    output_dir : Path
+        Directory where the output files will be written.
+    partial_global_mesh : PartialGlobalMesh
+        Structure containing the latitude and longitude grid.
+    partial_global_qualities : PartialGlobalQualities
+        Structure containing Vp, Vs, and Rho for all grid points.
+    lat_ind : int
+        Latitude index to determine the write mode (write or append).
+    min_vs : float, optional
+        Minimum Vs value to apply to the model.
+    logger : VMLogger
+        Logger instance for logging messages.
+
+    """
+    if logger is None:
+        logger = VMLogger("csv.wrote_global_qualities")
+
+    output_dir.mkdir(parents=True, exist_ok=True)
+    output_file = output_dir / "velocity_model.csv"
+
+    mode = "w" if lat_ind == 0 else "a"
+
+    if lat_ind == 0:
+        logger.log(f"Creating new CSV file: {output_file}", logger.INFO)
+
+    logger.log(f"Writing CSV output for latitude index {lat_ind}", logger.DEBUG)
+
+    try:
+        with open(output_file, mode, newline="") as csvfile:
+            writer = csv.writer(csvfile)
+
+            if lat_ind == 0:
+                writer.writerow(
+                    ["y", "x", "z", "lat", "lon", "depth", "vp", "vs", "rho", "inbasin"]
+                )
+
+            # Write data for this latitude slice
+            for k in range(partial_global_mesh.nx):  # Loop through longitude points
```

Comment:
Does this just only save a slice of the Velocity Model in latitude direction and all the depths? Is it meant to do all the longitude values as well

Reply:
For loop handing each latitude value is done in generate_velocity_model.py. ie. This write function in csv.py is called `ny` times with `lat_ind` argument. That is why we have a line to determine if this is a fresh write or append.
```
mode = "w" if lat_ind == 0 else "a"
```